.github/scripts/rust/lint.rs

```diff
@@ -0,0 +1,255 @@
+        if let Some(parent) = active.parent() {
+            active = parent;
+        } else {
+            break;
+        }
+    }
+
+    // we now apply them into a single configuration this allows for easy overwrites in child directories
+    files.reverse();
+
+    let mut lints = Lints::new();
+
+    for file in files {
+        let contents = fs::read_to_string(&file).expect(&format!("should be able to read {}", file.display()));
+
+        let file = toml::from_str(&contents).expect(&format!("should be valid toml ({})", file.display()));
+        lints.apply(file);
+    }
+
+    lints.into_file()
+}
+
+fn collect_cargo(cwd: &Path) -> LintFile {
+    let path = cwd.join(".cargo/config.toml");
+
+    if !path.exists() || !path.is_file() {
+        panic!(".cargo/config.toml needs to exist")
+    }
+
+    let contents = fs::read_to_string(&path).expect(&format!("should be able to read {}", path.display()));
+
+    // brute-force, search for the line:
+    // ## START CLIPPY LINTS ##
+    // and then:
+    // ## END CLIPPY LINTS ##
+    // take all values in between
+
+    if !contents.contains("## START CLIPPY LINTS ##") || !contents.contains("## END CLIPPY LINTS ##") {
+        panic!("malformed .cargo/config.toml, please add the required markers")
+    }
+
+    let file = contents.lines()
+        .map(|value| value.trim())
+        .skip_while(|value| *value != "## START CLIPPY LINTS ##")
+        .take_while(|value| *value != "## END CLIPPY LINTS ##")
+        .filter(|value| !value.is_empty())
+        .filter(|value| value.starts_with('"'))
+        // remove leading `"` and trailing `",`
+        .map(|value| &value[1..(value.len() - 2)]).fold(LintFile::default(), |mut acc, line| {
+            let (arg, lint) = line.split_at(2);
+
+            match arg {
+                "-A" => acc.allow.insert(lint.to_owned()),
+                "-W" => acc.warn.insert(lint.to_owned()),
+                "-D" => acc.deny.insert(lint.to_owned()),
+                "-F" => acc.forbid.insert(lint.to_owned()),
+                _ => panic!("unrecognized rust flag {arg}")
+            };
+
+            acc
+        });
+
+    file
+}
+
+fn apply(cwd: &Path) {
+    let path = cwd.join(".cargo/config.toml");
+
+    if !path.exists() || !path.is_file() {
+        panic!(".cargo/config.toml needs to exist")
+    }
+
+    let lints = collect_lints(cwd);
+
+    // read the `config.toml`
+    let contents = fs::read_to_string(&path).expect(&format!("should be able to read {}", path.display()));
+
+    if !contents.contains("## START CLIPPY LINTS ##") || !contents.contains("## END CLIPPY LINTS ##") {
+        panic!("malformed .cargo/config.toml, please add the required markers")
+    }
+
+    // strip out content between the markers
+    let prefix = contents.lines().take_while(|value| value.trim() != "## START CLIPPY LINTS ##");
+    let suffix = contents.lines().skip_while(|value| value.trim() != "## END CLIPPY LINTS ##");
+
+    let indent = "    ";
```

Comment:
We could use the same indent as `## START CLIPPY LINTS ##`

Reply:
fixed in [f798dce](https://github.com/hashintel/hash/pull/2075/commits/f798dce512d0598b591c5034af17ba792096a1fd)